Wikipedia/Code/FindInPageKeyboardBar.m

```diff
@@ -0,0 +1,88 @@
+#import "FindInPageKeyboardBar.h"
+#import "UIControl+BlocksKit.h"
+
+@interface FindInPageKeyboardBar()
+
+@property (strong, nonatomic) IBOutlet UIButton *closeButton;
+@property (strong, nonatomic) IBOutlet UIButton *clearButton;
+@property (strong, nonatomic) IBOutlet UIButton *previousButton;
+@property (strong, nonatomic) IBOutlet UIButton *nextButton;
+@property (strong, nonatomic) IBOutlet UIImageView *magnifyImageView;
+@property (strong, nonatomic) IBOutlet UILabel *currentMatchLabel;
+
+@end
+
+@implementation FindInPageKeyboardBar
+
+- (CGSize)intrinsicContentSize {
+    return CGSizeMake(UIViewNoIntrinsicMetric, 46);
+}
+
+- (void)dealloc {
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (instancetype)initWithCoder:(NSCoder *)coder{
+    self = [super initWithCoder:coder];
+    if (self) {
+        self.tintColor = [UIColor colorWithRed:0.3373 green:0.3373 blue:0.3373 alpha:1.0];
+        [[NSNotificationCenter defaultCenter] addObserver:self
```

Comment:
Usually the delegate is used here when you own the object. Normally only "outside" observers use notifications.